### PR TITLE
PWGHF: Add TPC/TOF nSig hist in Xic task

### PR DESCRIPTION
**Pull request by phymanshu**
In `PWGHF/D2H/Tasks/taskXic.cxx` task:
* Adding TOF/TPC nSigma histograms 
* Adding few more 1D and 2D histograms related to other observables
* Adding `Data` and `MC` directory structures to save histograms in the output

### CODE REVIEW

**Review by lvermunt — DISMISSED** [review]

**Comment by vkucera** [comment]
Hi @phymanshu @lvermunt , it is not clear to me why this PR is opened on top of #2585, which has a large overlap with this one, and why it has been approved. If it is supposed to be merged after #2585, it should be opened as a draft to clearly indicate that it is not ready for merging. Some comments in #2585 have not been addressed yet and are relevant for this PR as well. The title of this PR is not descriptive.

**Comment by lvermunt** [comment]
Hi @vkucera, my understanding is that this PR replaces #2585 (therefore also my comment in there to close that one). I don't know why @phymanshu chose to open a new PR instead of updating #2585, but since it addressed all your code-related comments (only the two new-line one around `struct HfTaskXic` is still open as far as I see) as well as mine, I approved.

**Comment by vkucera** [comment]
> 

OK, thanks for the clarification. Then please close the other PR and update this one to be a full standalone PR with a complete description without referring to the obsolete PR.

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 1]
```diff
@@ -34,172 +34,275 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
 struct HfTaskXic {
   Configurable<int> selectionFlagXic{"selectionFlagXic", 1, "Selection Flag for Xic"};
   Configurable<double> yCandMax{"yCandMax", -1., "max. cand. rapidity"};
+  Configurable<float> trackEtaCut{"trackEtaCut", 4.0, "eta cuts for tracks"};
+  Configurable<float> trackDcaXYCut{"trackDcaXYCut", 0.0025, "DCAxy cuts for tracks"};
+  Configurable<float> trackDcaZCut{"trackDcaZCut", 0.0025, "DCAz cuts for tracks"};
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:39]: Please, don't use `Cut`. Instead use `Min`, `Max` to clearly indicate the type of criterion and mention it in the description.
Also use the recommended order: `etaTrack`, `dcaXYTrack`,...

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 2]
```diff
@@ -34,172 +34,275 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
+    "registry", // histo not in pt bins
     {
-      {"hPtCand", "3-prong candidates;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
-      {"hPtRecSig", "3-prong candidates (matched);#it{p}_{T}^{rec.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
-      {"hPtRecBg", "3-prong candidates (unmatched);#it{p}_{T}^{rec.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
-      {"hPtGen", "MC particles (matched);#it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
-      {"hPtGenSig", "3-prong candidates (matched);#it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}} ///
+      {"Data/hPt", "3-prong candidates;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{360, 0., 36.}}}},      // pt Xic
+      {"Data/hEta", "3-prong candidates;candidate #it{eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},                    // eta Xic
+      {"Data/hPhi", "3-prong candidates;candidate #varphi;entries", {HistType::kTH1F, {{72, 0., 2. * constants::math::PI}}}}, // phi Xic
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:51]:
```suggestion
      {"Data/hPhi", "3-prong candidates;candidate #varphi;entries", {HistType::kTH1F, {{72, 0., constants::math::TwoPI}}}}, // phi Xic
```

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 3]
```diff
@@ -34,172 +34,275 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
-    registry.add("hCtRecBg", "3-prong candidates;proper lifetime (#Xi_{c}) * #it{c} (cm);;entries", {HistType::kTH2F, {{120, -20., 100.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
-    registry.add("hCPARecBg", "3-prong candidates;cosine of pointing angle;;entries", {HistType::kTH2F, {{110, -1.1, 1.1}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
-    registry.add("hEtaRecBg", "3-prong candidates;candidate #it{#eta};;entries", {HistType::kTH2F, {{100, -2., 2.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
-    registry.add("hPtProng0RecBg", "3-prong candidates;prong 0 #it{p}_{T} (GeV/#it{c});;entries", {HistType::kTH2F, {{100, 0., 10.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
-    registry.add("hPtProng1RecBg", "3-prong candidates;prong 1 #it{p}_{T} (GeV/#it{c});;entries", {HistType::kTH2F, {{100, 0., 10.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
-    registry.add("hPtProng2RecBg", "3-prong candidates;prong 2 #it{p}_{T} (GeV/#it{c});;entries", {HistType::kTH2F, {{100, 0., 10.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+
+    AxisSpec axisPidP = {100, 0.f, 10.0f, "#it{p} (GeV/#it{c})"};
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:66]:
```suggestion
    AxisSpec axisPPid = {100, 0.f, 10.0f, "#it{p} (GeV/#it{c})"};
```

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 4]
```diff
@@ -34,172 +34,275 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
+      } // rapidity selection
       if (yCandMax >= 0. && std::abs(yXic(candidate)) > yCandMax) {
         continue;
       }
 
-      auto mass = 0.;
-      auto massC = 0.;
+      auto massXicToPKPi = 0.;
+      auto massXicToPiKP = 0.;
+      auto candidatePt = candidate.pt();
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:251]:
```suggestion
      auto ptCandidate = candidate.pt();
```

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 5]
```diff
@@ -34,172 +34,275 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
+      registry.fill(HIST("Data/hDecLenErrVsPt"), candidate.errorDecayLength(), candidatePt);
+      registry.fill(HIST("Data/hChi2PCAVsPt"), candidate.chi2PCA(), candidatePt);
+
+      const auto& trackProng0 = candidate.prong0_as<aod::BigTracksPID>(); // bachelor track
+      const auto& trackProng1 = candidate.prong1_as<aod::BigTracksPID>(); // bachelor track
+      const auto& trackProng2 = candidate.prong2_as<aod::BigTracksPID>(); // bachelor track
+
+      auto prong0Momentum = trackProng0.p();
+      auto prong1Momentum = trackProng1.p();
+      auto prong2Momentum = trackProng2.p();
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:205]:
```suggestion
      auto momentumProng0 = trackProng0.p();
      auto momentumProng1 = trackProng1.p();
      auto momentumProng2 = trackProng2.p();
```

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 6]
```diff
@@ -34,172 +34,275 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
+        if (std::abs(track.dcaXY()) > trackDcaXYCut || std::abs(track.dcaZ()) > trackDcaZCut) { // hardcoded cut TO BE REFINED
+          continue;
+        }
+        nTracks++;
+      }
+    }
+
+    registry.fill(HIST("Data/hMultiplicity"), nTracks); // filling the histo for multiplicity
+    for (auto const& candidate : candidates) {
+      auto candidatePt = candidate.pt();
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:161]:
```suggestion
      auto ptCandidate = candidate.pt();
```

File: PWGHF/D2H/Tasks/taskXic.cxx
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:1]: - Please fix the position of the line `#include "Framework/runDataProcessing.h"`
- Add yourself on the list of authors.
- Fix missing `const` in the `for` loops.

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
Thanks @phymanshu  for addressing the comments. The PR looks fine for me now. Just please give your newly added commits better names when you are addressing the comments.

**Pull request merged**